internal/datanode/importv2/scheduler.go

```diff
@@ -126,33 +131,42 @@ func WrapLogFields(task Task, fields ...zap.Field) []zap.Field {
 	return res
 }
 
-func (e *executor) handleErr(task Task, err error, msg string) {
+func tryFreeFutures(futures map[int64][]*conc.Future[any]) {
+	for k, fs := range futures {
+		fs = lo.Filter(fs, func(f *conc.Future[any], _ int) bool {
+			return !f.Done()
```

Comment:
no need to call f.Await() for futures which is done ?

Reply:
done